sklearn/ensemble/weight_boosting.py

```diff
@@ -748,6 +749,24 @@ class in ``classes_``, respectively.
             else:
                 yield pred / norm
 
+    @staticmethod
+    def _compute_proba_from_decision(decision, n_classes):
+        """Compute probabilities from the decision function.
+
+        This is based eq. (4) of [1] where:
+            p(y=c|X) = exp((1 / K-1) f_c(X)) / sum_k(exp((1 / K-1) f_k(X)))
+                     = softmax((1 / K-1) * f(X))
+
+        References
+        ----------
+        .. [1] J. Zhu, H. Zou, S. Rosset, T. Hastie, "Multi-class AdaBoost",
+               2009.
+        """
+        if n_classes == 2:
+            decision = np.vstack([-decision, decision]).T
```

Comment:
can you explain this? is it in the paper too?

Reply:
My original thought was that we were keeping one of the 2 columns when computing the decision function. Actually, we do something a bit different:

```python
        if n_classes == 2:
            pred[:, 0] *= -1
            return pred.sum(axis=1)
```

You originally have symmetry between both classes. However, I should divide decision by 2 since that we are summing.